velox/common/memory/SharedArbitrator.cpp

```diff
@@ -258,6 +258,31 @@ SharedArbitrator::SharedArbitrator(const Config& config)
                       << participantConfig_.toString();
 }
 
+void SharedArbitrator::shutdown() {
+  {
+    std::lock_guard<std::mutex> l(stateLock_);
+    VELOX_CHECK(globalArbitrationWaiters_.empty());
+    if (shutdown_) {
+      return;
+    }
```

Comment:
Do we allow shutdown() to be called multiple times?

Reply:
It will call only once in production and it is re-entry to ease testing